tendrl/ceph_integration/objects/pool/atoms/create/__init__.py

```diff
@@ -1,18 +1,27 @@
 from tendrl.ceph_integration.manager.crud import Crud
-from tendrl.ceph_integration.manager import utils as manager_utils
 from tendrl.ceph_integration import objects
+from tendrl.ceph_integration.objects.pool import Pool
 
 
 class Create(objects.CephIntegrationBaseAtom):
-    def __init__(self, config=None, *args, **kwargs):
+    obj = Pool
+    def __init__(self, *args, **kwargs):
         super(Create, self).__init__(*args, **kwargs)
-        self.obj = objects.Pool
 
-    def run(self, parameters):
-        fsid = manager_utils.get_fsid()
-        attrs = dict(name=parameters['Pool.poolname'],
-                     pg_num=parameters['Pool.pg_num'],
-                     min_size=parameters['Pool.min_size'])
-        crud = Crud(parameters['manager'])
-        crud.create(fsid, "pool", attrs)
+    def run(self):
+        attrs = dict(name=self.parameters['Pool.poolname'],
+                     pg_num=self.parameters['Pool.pg_num'],
+                     min_size=self.parameters['Pool.min_size'],
+                     type=self.parameters.get('Pool.type'),
+                     erasure_code_profile=self.parameters.get(
+                         'Pool.erasure_code_profile'
+                     ))
+        if 'Pool.quota_enabled' in self.parameters and \
+            self.parameters['Pool.quota_enabled'] == True:
+            attrs['quota_max_objects'] = \
+                self.parameters.get('Pool.quota_max_objects')
+            attrs['quota_max_bytes'] = \
+                self.parameters.get('Pool.quota_max_bytes')
+        crud = Crud()
+        crud.create("pool", attrs)
```

Comment:
Let this be as is, We need a better strategy to trigger syncs, @shtripat you can modify the crud.create to sync object on success